### PR TITLE
App config

### PR DESCRIPTION
**Pull request by fernandatoledo**
#### Description:

* Add useful packages and the api communication files.

---

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1 @@
+API_URL=https://api-example.com
```
Comment (nsantos16) [inline .env]: Add `.env` to .gitignore

Comment (fernandatoledo) [reply .env]: Thank you, totally missed that. b02f98b7ca74dc647f884ebe99dbc579d01e51cf

**Comment by fernandatoledo** [comment]
> 👌

@juanlacruz 01c16ef2b6fecefba37a7ac3d3a3bfda32bccc61 😉

File: android/app/src/main/java/com/targetmvd/MainApplication.java [hunk 2]
```diff
@@ -7,6 +7,8 @@
 import com.facebook.hermes.reactexecutor.HermesExecutorFactory;
 import com.facebook.react.bridge.JavaScriptExecutorFactory;
 import com.facebook.react.ReactApplication;
+import com.lugg.ReactNativeConfig.ReactNativeConfigPackage;
+import com.reactnativecommunity.asyncstorage.AsyncStoragePackage;
```
Comment (mato2593) [inline android/app/src/main/java/com/targetmvd/MainApplication.java]: I think you can remove these two imports

File: src/App.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import { createStore, applyMiddleware } from 'redux'
+import ReduxThunk from 'redux-thunk'
+import { sessionService } from 'redux-react-native-session'
+import reducers from './reducers'
+
+class App extends Component {
+
+  render() {
+    const store = createStore(reducers, {}, applyMiddleware(ReduxThunk))
+    sessionService.initSessionService(store)
```
Comment (mato2593) [inline src/App.js]: Is it ok to do this for every render?

File: src/api/utils/handleErrors.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    throw new Error({ message: 'No response returned from fetch' })
+  }
+
+  if (response.ok) {
+    await saveSessionHeaders(response.headers)
+    return response
+  }
+
+  try {
+    await sessionService.loadSession()
```
Comment (mato2593) [inline src/api/utils/handleErrors.js]: Why do you need to load the session here?

File: src/actions/AuthActions.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    dispatch(loginSuccess())
+  } catch (err) {
+    dispatch(loginFail())
+  }
+}
+
+const loginFail = (dispatch) => {
+  console.log('dispatch fail')
+  dispatch({ type: types.LOGIN_FAIL })
+}
```
Comment (mato2593) [inline src/actions/AuthActions.js]:
```suggestion
const loginFail = () => ({
  type: types.LOGIN_FAIL 
})
```

File: src/actions/AuthActions.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+const loginSuccess = (dispatch, user) => {
+  console.log('dispatch success')
+  dispatch({ type: types.LOGIN_SUCCESS, payload: user })
+  Actions.main()
```
Comment (mato2593) [inline src/actions/AuthActions.js]: The same applies to this action. Also, what is this `Actions.main()`?

File: android/app/src/main/java/com/targetmvd/MainApplication.java (hunk 2 above)
Comment (fernandatoledo) [reply android/app/src/main/java/com/targetmvd/MainApplication.java]: It's not something I added manually, I don't think I should modify something that was added while installing a module, should I remove it anyway?

File: src/actions/AuthActions.js (hunk 5 above)
Comment (fernandatoledo) [reply src/actions/AuthActions.js]: This file should not even be part of the PR, I'll delete it 🙈

File: src/actions/AuthActions.js (hunk 6 above)
Comment (fernandatoledo) [reply src/actions/AuthActions.js]: This file should not even be part of the PR, I'll delete it 🙈

File: android/app/src/main/java/com/targetmvd/MainApplication.java (hunk 2 above)
Comment (mato2593) [reply android/app/src/main/java/com/targetmvd/MainApplication.java]: Mmm, that's a good point. I mean, those imports are not required, but there are previous imports that weren't being used, so just leave them.

File: src/api/utils/handleErrors.js (hunk 4 above)
Comment (mato2593) [reply src/api/utils/handleErrors.js]: Nevermind, it's needed to skip the status validation.

File: src/App.js (hunk 3 above)
Comment (fernandatoledo) [reply src/App.js]: It's definitely not, you can check the change here 83aec94a94cd2cd3a2afa7e2b96eb1b0609c5d99 :)

**Pull request merged**